semantic-scope-graph/src/Control/Carrier/Sketch/Fresh.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+module Control.Carrier.Sketch.Fresh
```

Comment:
I’m not sure this is the best name for the module; what other carriers are you envisioning existing?

Corollary to which: what motivates this being modelled as an effect? Would it be better to model it e.g. using `Fresh` directly?

Reply:
As I mentioned on the call, we’re still exploring this space. For now I’m inclined to leave it as an effect, but I don’t think it will be too onerous to switch back to the old style if we want to.

Reply:
We’ll definitely drop the `.Fresh`.